openssl/src/symm.rs

```diff
@@ -626,10 +626,8 @@ impl Crypter {
 
         ctx.set_key_length(key.len())?;
 
-        if let (Some(iv), Some(iv_len)) = (iv, t.iv_len()) {
-            if iv.len() != iv_len {
-                ctx.set_iv_length(iv.len())?;
-            }
+        if let (Some(iv), Some(_iv_len)) = (iv, t.iv_len()) {
+            ctx.set_iv_length(iv.len())?;
```

Comment:
What do we think about this instead? I reintroduced the length check and added specific handling for the AES CCM ciphers, which are the ones with the IV length issue.

```suggestion
        if let (Some(iv), Some(iv_len)) = (iv, t.iv_len()) {
            if iv.len() != iv_len
                || matches!(
                    t.nid(),
                    Nid::AES_128_CCM | Nid::AES_192_CCM | Nid::AES_256_CCM
                )
            {
                ctx.set_iv_length(iv.len())?;
            }
```

Reply:
Looks like boringssl's not happy with it though.

Reply:
Also, agree that it's not very pretty but it should (hopefully!) get the job done!